core/src/main/java/org/infinispan/distexec/DefaultExecutorService.java

```diff
@@ -115,7 +116,14 @@
     *           Cache node initiating distributed task
     */
    public DefaultExecutorService(Cache<?, ?> masterCacheNode) {
-      this(masterCacheNode, Executors.newSingleThreadExecutor(), true);
+      this(masterCacheNode, createLocalExecutor(masterCacheNode), true);
+   }
+
+   public static ExecutorService createLocalExecutor(Cache<?, ?> masterCacheNode) {
+      String nodeName = masterCacheNode != null ? masterCacheNode.getCacheManager().getCacheManagerConfiguration().transport().nodeName() : "";
```

Comment:
Should this be null if cache is null? Just want to keep it consistent with other below.